### PR TITLE
codeintel: Add CalculateVisibleUploads to store

### PR DESCRIPTION
**Pull request by efritz**
This PR adds methods to the store that populate/read some of the nearest upload data in the new tables defined in [RFC 186](https://docs.google.com/document/d/1NiAQG4muTljgk8qirYs2aN5Zt3lfk4lgwNqOm9WOH4s).

This is part of a larger effort towards https://github.com/sourcegraph/sourcegraph/issues/12098.

### CODE REVIEW

File: enterprise/internal/codeintel/store/commits.go [hunk 1]
```diff
@@ -137,3 +164,170 @@ func diff(left, right []string) []string {
...
+
+	// Insert new data for this repository in batches - it's likely we'll exceed the maximum
+	// number of placeholders per query so we need to break it into several queries below this
+	// size.
+	for _, batch := range batchQueries(nearestUploadsRows, MaxPostgresNumParameters/4) {
+		if err := tx.queryForEffect(ctx, sqlf.Sprintf(
+			`INSERT INTO lsif_nearest_uploads (repository_id, "commit", upload_id, distance) VALUES %s`,
+			sqlf.Join(batch, ","),
+		)); err != nil {
+			return err
```
Comment (uwedeportivo) [inline enterprise/internal/codeintel/store/commits.go:273]: so i'm confused here by this err variable which is local to this scope and then the defer function.  you want the transaction cancelled. that closure captures the top-level err var, but not this one, correct ?

File: enterprise/internal/codeintel/store/commits_test.go [hunk 2]
```diff
@@ -202,3 +203,366 @@ func TestUpdateCommitsNoUnknownData(t *testing.T) {
 func strPtr(v string) *string {
 	return &v
 }
+
+func TestMarkRepositoryAsDirty(t *testing.T) {
```
Comment (uwedeportivo) [inline enterprise/internal/codeintel/store/commits_test.go:207]: super nice tests !

File: enterprise/internal/codeintel/store/commits.go (hunk 1 above)
Comment (efritz) [reply enterprise/internal/codeintel/store/commits.go:273]: The top-level err value (the one that is named as part of the return value) is going to have whatever value is returned from the function, regardless of it's a return of a variable in a sub-scope.

Comment (efritz) [reply enterprise/internal/codeintel/store/commits.go:273]: Thanks - I see that I didn't name the return value. Fixing!

**Pull request merged**